.github/workflows/base_tests.yml

```diff
@@ -0,0 +1,130 @@
+name: Base Tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  workflow_call:
+    inputs:
+      include:
+        description: 'Include matrix configurations'
+        required: false
+        type: string
+        default: "[]"
+      exclude:
+        description: 'Exclude matrix configurations'
+        required: false
+        type: string
+        default: "[]"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  delete_cache:
```

Comment:
Could you please elaborate more on why this is necessary.

Reply:
That's something that exists in the current `daily_common.yml` workflow.

What it does is `Deletes GitHub Actions caches for a branch.`. I think its utility is better described with the example the repository provides: `Caches created in the HEAD branch of a pull request are no longer used, so delete them immediately to save repository cache space.`.

I think it still provides some utility and, although it isn't essential, it might be a good idea to keep it on a different workflow. That way it could be run only on a `PR Closed` event, as they show in their example.

What do you think?

REF: https://github.com/snnaplab/delete-branch-cache-action